torchgeo/trainers/autoregression.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Trainers for autoregression."""
+
+from typing import Any
+
+import torch.nn as nn
+from torch import Tensor
+from torchmetrics import MetricCollection
+from torchmetrics.regression import MeanAbsoluteError, MeanSquaredError
+
+from torchgeo.models import Seq2Seq
+
+from .base import BaseTask
+
+
+class AutoregressionTask(BaseTask):
+    """Autoregression."""
+
+    def __init__(
+        self,
+        model: str = 'seq2seq',
+        input_size: int = 1,
+        input_size_decoder: int = 1,
+        output_size: int = 1,
+        target_indices: list[int] | None = None,
+        loss: str = 'mse',
+        lr: float = 1e-3,
+        patience: int = 10,
+        **kwargs: dict[str, Any],
+    ) -> None:
+        """Initialize a new AutoregressionTask instance.
+
+        Args:
+            model: Name of the model to use, currently supports 'seq2seq'.
+                Defaults to 'seq2seq'.
+            input_size: The number of features in the input. Defaults to 1.
+            input_size_decoder: The number of features in the decoder input.
+                Defaults to 1.
+            output_size: The number of features output by the model. Defaults to 1.
+            target_indices: The indices of the target(s) in the dataset. If None, uses all features. Defaults to None.
+            loss: One of 'mse' or 'mae'. Defaults to 'mse'.
+            lr: Learning rate for optimizer. Defaults to 1e-3.
+            patience: Patience for learning rate scheduler. Defaults to 10.
```

Comment:
The default values and valid options will be documented in the function signature, no need to duplicate in the docstring. For example, see https://torchgeo.readthedocs.io/en/stable/api/trainers.html#torchgeo.trainers.ClassificationTask.__init__